.github/workflows/build-macos.yaml

```diff
@@ -0,0 +1,35 @@
+name: Build on Mac OS
+run-name: Build on Mac OS
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    paths:
+      - '**.go'
+      - 'go.mod'
+      - 'go.sum'
+      - '**.rs'
+      - 'Cargo.toml'
+      - 'Cargo.lock'
+
+jobs:
+  build:
+    name: Build all tools on Mac OS
+    runs-on: macos-12 # TODO(r0mant): Update with large runner when it's available
+
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout Teleport
+        uses: actions/checkout@v3
+
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
```

Comment:
Similar to the last PR - any way to determine this automatically so we don't have to keep it in sync with the makefile?

Reply:
Maybe? I wonder if we can make a `make go-version` Makefile target that would only output Go version and set it to a variable in this workflow. Let me take a stab at it.